eclair-core/src/main/scala/fr/acinq/eclair/channel/Commitments.scala

```diff
@@ -71,12 +72,14 @@ case class Commitments(localParams: LocalParams, remoteParams: RemoteParams,
 
   def addRemoteProposal(proposal: UpdateMessage): Commitments = Commitments.addRemoteProposal(this, proposal)
 
-  def announceChannel: Boolean = (channelFlags & 0x01) != 0
+  val announceChannel: Boolean = (channelFlags & 0x01) != 0
 
-  def availableBalanceForSendMsat: Long = {
+  def usableBalances: UsableBalances = {
     val reduced = CommitmentSpec.reduce(remoteCommit.spec, remoteChanges.acked, localChanges.proposed)
-    val feesMsat = if (localParams.isFunder) Transactions.commitTxFee(Satoshi(remoteParams.dustLimitSatoshis), reduced).amount * 1000 else 0
-    reduced.toRemoteMsat - remoteParams.channelReserveSatoshis * 1000 - feesMsat
+    val commitTxFee = Transactions.commitTxFee(Satoshi(remoteParams.dustLimitSatoshis), reduced).amount * 1000
+    val localBalance = reduced.toRemoteMsat - remoteParams.channelReserveSatoshis * 1000 - { if (localParams.isFunder) commitTxFee else 0 }
```

Comment:
`localBalance` needs to be computed from the local commitment (see the difference ine the reduced commitment in `sendAdd` and `receiveAdd`).

How about:
1) keep `availableBalanceForSendMsat` as it was
2) add an `availableBalanceForReceiveMsat` that is symmetrical (it doesn't duplicate code because the commitment is computed differently)
3) define the `UsableBalance` class in `Relayer`, and make it contain the `isFunder` boolean and `remoteNodeId` and `channel_update`, so that we don't mix concerns.